### ISSUE TITLE
epo file format

### ISSUE THREAD

**Issue by xinkaitong** [grenaud/glactools#25]
hello, grenaud! now, I just have an outgroup vcf file (other mammals not human), could you tell me how can I transform vcf file to epo file?

**Comment by grenaud** [comment]
That is very tough! What is much easier is to transform that VCF to
ACF, and use the specific population as the root/anc. However be wary that
you don't really have an ancestor, you have an outgroup :-)

**Comment by xixifa** [comment]
Hello,
If I want to calculate β2 using an outgroup to call substitutions and ancestral alleles, can I just get epo files from outgroup vcf file, without common ancestor?
Thank you very much

**Comment by grenaud** [comment]
Yes! You need to convert VCF to ACF and run replaceancestor I believe. Let me know if that works!

**Comment by xixifa** [comment]
Thank you. I will try it ! 
If I use vcfm2acf to convert VCF to ACF , then I use usepopsrootanc to tell the root(outgroup), finally I use the replaceanc,  the resulting file is the epo file that I need in betascan2 software? 
Does this epo file include all the individuals I calculate later?

**Comment by xixifa** [comment]
Hello, I have another question.
Is the file2.acf.gz in https://github.com/grenaud/glactools the result using the glactools with vcfm2acf --onlyGT --fai parameter?
I am not sure if using these parameters is correct?

Thanks

**Comment by xixifa** [comment]
Hello, when I use the "usepopsrootanc" parameter, the error is as follows
![image](https://github.com/grenaud/glactools/assets/76728696/8dd8b194-6747-459d-a246-3051eb9c13bd)

 I think it lack ancestral information, it is right?

**Comment by xixifa** [comment]
Hello, 
the file1.acf.gz only have a outgroup individual:
![image](https://github.com/grenaud/glactools/assets/76728696/61930e93-0364-4bde-887a-4d8c2c64f8ce)
when I use the "usepopsrootanc" parameter, I only tell the outgroup, then the error is as follows
![image](https://github.com/grenaud/glactools/assets/76728696/8b1340ad-87fa-403e-85d7-02f969f87dd2)
I think it lack ancestral information, it is right? so the file1.acf.gz needs the root and ancestor information?

**Comment by xixifa** [comment]
Hello,
What does "ChimpHumanAncestor " refer to specifically?

Thank you very much

**Comment by grenaud** [comment]
> Thank you. I will try it ! If I use vcfm2acf to convert VCF to ACF , then I use usepopsrootanc to tell the root(outgroup), finally I use the replaceanc, the resulting file is the epo file that I need in betascan2 software? Does this epo file include all the individuals I calculate later?

Would you describe seems like it will work. The EPO is only human+chimp+other great apes and the inferred ancestral states, there is a single human "individual" and that is the reference

**Comment by grenaud** [comment]
> Hello, I have another question. Is the file2.acf.gz in https://github.com/grenaud/glactools the result using the glactools with vcfm2acf --onlyGT --fai parameter? I am not sure if using these parameters is correct?
> 
> Thanks

I am not sure how this was generated but acf files can be generated in very different ways.

**Comment by grenaud** [comment]
There is root and ancestor. In the case of humans, we can think of the chimps as the "root" as all humans are equitistant to the chimp. However there is a chimp-human ancestor, we do not have access to the sequence of its genome as it died out 6 million years ago. However we can make a probabilistic inference of what this genome had in terms of alleles. This is precisely what EPO does.

**Comment by xixifa** [comment]
Hello,
May I ask if I have only root and ancestor, but no groups like other apes, is it OK to generate file3.acf.gz as my epo file for Betascan2？
![image](https://github.com/grenaud/glactools/assets/76728696/98c3fa28-2b48-4bca-b13d-e8cf5130245e)
Thank you very much

**Comment by xixifa** [comment]
Hello,
Thank you for your reply!
1. May I ask if I have only root and ancestor, but no groups like other apes, is it OK to generate the follow file.acf.gz as my epo file for Betascan2？
![image](https://github.com/grenaud/glactools/assets/76728696/5f498aeb-e4d6-498a-9cf8-4279e6adc639)
2. After I get the epo file, I get an error adding the index file? How to create this index file?
![image](https://github.com/grenaud/glactools/assets/76728696/b56fa66c-d277-411a-9a84-29f135ccffdf)

3. The first line of the epo file seems to be different from the example you showed. Is it correct for my epo file?
![image](https://github.com/grenaud/glactools/assets/76728696/e42ccd1f-7d05-4154-b2f0-f45caba134af)
Thanks

**Comment by grenaud** [comment]
How did you generate this file? Could you paste the command lines?

**Comment by xixifa** [comment]
I think I misunderstood your meaning.
Is file4.acf.gz in your example (glactools replaceanc  file2.acf.gz  file3.acf.gz  > file4.acf.gz) equal to the 1kg_chr6.AFRsubset.acf.gz in following code ( glactools vcfm2acf --onlyGT --epo all.epo.gz --fai human_g1k_v37.fasta.fai chr6.AFRsubset.recode.vcf > 1kg_chr6.AFRsubset.acf.gz)?
These two files are actually equivalent, right?

**Comment by xixifa** [comment]
sorry，
1. glactools replaceanc  file2.acf.gz  <(glactools usepopsrootanc -u file1.acf.gz chimp ChimpHumanAncestor )  > file4.acf.gz 
2. glactools vcfm2acf --onlyGT --epo all.epo.gz --fai human_g1k_v37.fasta.fai chr6.AFRsubset.recode.vcf > 1kg_chr6.AFRsubset.acf.gz

What I understand is that if there is no epo file, file4 generated according to step 1 and 1kg_chr6.AFRsubset.acf.gz generated using epo in 2 are both acf files containing outgroup information.  steps 1 and 2 can generate the same file, and the steps are equivalent, is it right?

**Comment by grenaud** [comment]
While I encourage people to use file descriptors, they are not great for debugging because you don't know which one fails.

Could you try to break down these commands into three separate ones, produce the output for each and inspect that using glactools view?

**Comment by xixifa** [comment]
Yes, I used 3 commands to generate the file successfully.

Since my species is not human, I want to know if the output file of 1 is the equivalent of the output file of 2 ?

Thank you very much

**Comment by grenaud** [comment]
Could you please paste the commands? I am not sure what is file 1 and file 2 here.

**Comment by xixifa** [comment]
![image](https://github.com/grenaud/glactools/assets/76728696/9763f35e-cfc7-463d-88ec-a31440977d0b)
I mean these two ways above. Since my object is not human, there is no epo file, so I can only use the method of 1. I am not sure whether the file I generated is the input file of betascan2? the following is my output
![image](https://github.com/grenaud/glactools/assets/76728696/eeb44d56-d943-486b-9cd1-9d8904e40dff)

**Comment by grenaud** [comment]
Again, please no file descriptors, just single commands.

**Comment by xixifa** [comment]
The following is code of my test file.
IN1 is outgroup, ZA1  is ancestor. 
test.chr24.vcf is my pop1 file.

glactools vcfm2acf --onlyGT --fai $ref_genome  outgroup_root_test.chr24.vcf > outgroup_root_test.chr24.acf.gz
glactools vcfm2acf --onlyGT --fai $ref_genome test.chr24.vcf > test.chr24.acf.gz
glactools usepopsrootanc outgroup_root_test.chr24.acf.gz IN1 ZA1 > outgroup_root_test_identify.chr24.acf.gz
glactools replaceanc test.chr24.acf.gz outgroup_root_test_identify.chr24.acf.gz > outgroup_root_test_merge.chr24.acf.gz

**Comment by grenaud** [comment]
ok thank you! which command fails?

**Comment by xixifa** [comment]
No error reported.
I'm just not sure if the outgroup_root_test_merge.chr24.acf.gz is the betascan2 input file?

**Comment by grenaud** [comment]
ah no it is not! you need to use glactools acf2betascan

**Comment by xixifa** [comment]
glactools acf2betascan --useroot  outgroup_root_test_merge.chr24.acf.gz | gzip > outgroup_root_test.chr24.beta.txt.gz

Is it right to do ?

**Comment by grenaud** [comment]
should be!

**Comment by xixifa** [comment]
Ok! Thank you very much !
